core_types/build.rs

```diff
@@ -1,5 +1,8 @@
-use std::process::Command;
+use std::{path::Path, process::Command};
 fn main() {
+    if Path::new("../.git/HEAD").exists() {
```

Comment:
what's this?

Reply:
Can we use pathbuf instead? Just handles the seperators better

Reply:
Isn't Path just a slice of a PathBuf?

Reply:
Pathbuf is to string as Path is to &str , so what we would want is to PathBuf::().join() so that we dont have to handle the seperator logic for different Oses

Reply:
@maackle @AshantiMutinta yes and this is my code and i hate it whoops